__tests__/Unit/hooks/useUserData.test.ts

```diff
@@ -0,0 +1,53 @@
+import { renderHook } from '@testing-library/react-hooks';
+import useUserData from '@/hooks/useUserData';
+import { useGetUserQuery } from '@/app/services/userApi';
+
+jest.mock('@/app/services/userApi');
+
+describe('useUserData', () => {
+    beforeEach(() => {
+        jest.resetAllMocks();
+    });
+
+    test('should return authorization status as true', () => {
+        const mockData = {
+            roles: {
+                admin: true,
+                super_user: true,
+            },
+        };
+
+        (useGetUserQuery as jest.Mock).mockReturnValue({
```

Comment:
You wouldn't need to mock an RTK hook as we are using msw in the project, so the API calls are mocked using that, please check on this once

Reply:
I will look into it . Thanks!